var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -84,7 +85,12 @@ def determine_version(cls, lib):
     @property
     def libs(self):
         shared = "+shared" in self.spec
-        return find_libraries(["libz"], root=self.prefix, recursive=True, shared=shared)
+        libnames = ["libz"]
+        if sys.platform == "win32":
+            libnames.append("zdll" if shared else "zlib")
```

Comment:
(question)

You mention:

> Zlib is named either zdll or zlib depending on the library type (or both if shared)

when you say "both if shared" is one of them a symlink or something?

Reply:
zlib is a library that produces a static build and a shared build at once. Typically this results in a naming collision due to the import and static library having the same name and extension and I have to either disable one or the other or patch the CMake to make it work, but Zlib did the right thing up front and re-named the dll.